src/stores/mainVehicle.ts

```diff
@@ -125,29 +162,52 @@ export const useMainVehicleStore = defineStore('main-vehicle', () => {
 
   /**
    * Arm the vehicle
+   * @returns { void } A Promise that resolves when arming is successful or rejects if an error occurs or the action is cancelled.
    */
-  function arm(): void {
-    mainVehicle.value?.arm()
+  function arm(): Promise<void> {
+    return slideToConfirm(() => {
+      if (!mainVehicle.value) {
+        throw new Error('action rejected or failed')
+      }
+      mainVehicle.value.arm()
+    })
```

Comment:
Although this logic is fine and we can proceed with it, I have a suggestion of a slight different one that seems to be less intrusive in the code:

```typescript
async function arm(): Promise<void> {
  if (!mainVehicle.value) throw new Error('No vehicle available to be armed.')
  
  const slideConfirmed = await slideToConfirm('Slide to arm', 10000)
  if (slideConfirmed) {
    mainVehicle.value.arm()
    return
  }
  console.error('No confirmation for arming action received') 
}
```

This way we don't have to pass the whole success code (in this case it's only `mainVehicle.value.arm()`, but on more complex cases could be several lines of code) to the confirmation slider, but just to read its result. This way the code get's cleaner, more readable, with less indentations, and its more in line [with what we currently use for SweetAlert2 confirmation dialogs](https://github.com/bluerobotics/cockpit/blob/f8d3f981a90d89fcc4002e5604ba86a19935d483/src/components/EditMenu.vue#L357), for example.

Reply:
this also makes sense. I will change. Thanks for all the great feedback!

Reply:
I forgot to do this. let me know if you would like me to make that change. If I do, do you think I should reset that commit make the change and recommit, or would it be better to make a separate commit to make that change?

Reply:
I think it would be cool, but at the same time it would be an improvement, so you can add it to a following commit, so you don't have to recommit everything, or we can settle with how it is right now and allow this part to be improved on a future PR, without hurries. I'm fine with both, so let me know what you prefer.

Reply:
After giving it another look, I think it would take me a little bit of time to refactor. 

If you are ok with it I think I would like to leave it how it is for now. :slightly_smiling_face:

Reply:
No problem!
I will test it tomorrow morning so we can merge.